app/code/core/Mage/GoogleAnalytics/Block/Ga.php

```diff
@@ -280,6 +280,161 @@ protected function _getOrdersTrackingCodeAnalytics()
         return implode("\n", $result);
     }
 
+    /**
+     * Render Enhanced eCommerce for GA4
+     * @return string
+     * @throws Mage_Core_Model_Store_Exception
+     */
+    protected function _getEnhancedECommCodeAnalytics4()
+    {
+        $helper = $this->helper('googleanalytics');
+        if (!$helper->isEnhancedECommEnabled()) {
+            return '';
+        }
+
+        $result = [];
+
+        //product page
+        if ($this->getRequest()->getModuleName() == 'catalog' && $this->getRequest()->getControllerName() == 'product') {
+            $productViewed = Mage::registry('current_product');
+            $eventData = [];
+            $eventData['currency'] = Mage::app()->getStore()->getCurrentCurrencyCode();
+            $eventData['value'] = number_format($productViewed->getFinalPrice(), 2);
+            $eventData['items'] = [];
+            $eventData['items'][] = [
+                'id' => $productViewed->getSku(),
+                'name' => $productViewed->getName(),
+                'list_name' => 'Product Detail Page',
+                'brand' => $productViewed->getAttributeText('manufacturer'),
+                'category' => 'Products',
+                'price' => number_format($productViewed->getFinalPrice(), 2),
+            ];
+
+            $result[] = "gtag('event', 'view_item', " . json_encode($eventData, JSON_THROW_ON_ERROR) . ");";
+        }
+
+        //category page
+        elseif ($this->getRequest()->getModuleName() == 'catalog' && $this->getRequest()->getControllerName() == 'category') {
+            $layer = Mage::getSingleton('catalog/layer');
+            $category = $layer->getCurrentCategory();
+            $productCollection = $layer->getProductCollection();
+            $toolbarBlock = Mage::app()->getLayout()->getBlock('product_list_toolbar');
+            $pageSize = $toolbarBlock->getLimit();
+            $currentPage = $toolbarBlock->getCurrentPage();
+            if ($pageSize !== 'all') {
+                $productCollection->setPageSize($pageSize)->setCurPage($currentPage);
+            }
+            $eventData = [];
+            $eventData['currency'] = Mage::app()->getStore()->getCurrentCurrencyCode();
+            $eventData['value'] = 0.00;
```

Comment:
```suggestion
            $eventData['value'] = '0.00';
```

Reply:
@fballiano This is already set to a string after the total value up in the loop. 
```
$eventData['value'] = number_format($eventData['value'], 2);
```